app/src/main/java/de/danoeh/antennapod/fragment/FeedSettingsFragment.java

```diff
@@ -256,6 +257,7 @@ protected void onConfirmed(String username, String password) {
                         feedPreferences.setUsername(username);
                         feedPreferences.setPassword(password);
                         DBWriter.setFeedPreferences(feedPreferences);
+                        DBTasks.forceRefreshFeed(requireContext(), feed, true);
```

Comment:
`setFeedPreferences` is an asynchronous operation. Doing it like this might refresh before the updated credentials actually made it to the database.

Reply:
Great point. Thank you for providing great guidance - always intimidating getting involved with a new project!

With `setFeedPreferences` returning a `Future`, we need to wait until it is completed before refreshing the feed.

Thoughts on waiting for the future to be done by checking its status every 200ms?
```java
 @Override
protected void onConfirmed(String username, String password) {
  feedPreferences.setUsername(username);
  feedPreferences.setPassword(password);
  Future<?> setPreferencesFuture = DBWriter.setFeedPreferences(feedPreferences);
  
  while (true) {
      if (setPreferencesFuture.isDone()) {
          DBTasks.forceRefreshFeed(requireContext(), feed, true);
          break;
      } else if (setPreferencesFuture.isCancelled()) {
          Log.d(TAG, "Setting preferences with updated credentials cancelled.");
          break;
      }
  
      try {
          Thread.sleep(200);
      } catch (InterruptedException e) {
          Log.d(TAG, "Thread was interrupted while waiting. Feed not auto refreshed after credential change.");
          e.printStackTrace();
          break;
      }
  }
}
```

Reply:
You can just call `get()` on that future to wait until it is done. But I think that should then be done in a background thread. Otherwise it blocks the UI.